onnxruntime/core/optimizer/transpose_optimizer/optimizer_api_impl.cc

```diff
@@ -909,7 +909,18 @@ Status TransformLayoutForEP(Graph& graph, bool& modified, const IExecutionProvid
         onnx_layout_transformation::WrapTransposesAroundNode(*api_graph, *node, {&input_perm}, {&output_perm});
       }
 
-      onnx_layout_transformation::SwapNodeOpTypeAndDomain(*api_graph, *node, node->OpType(), kMSInternalNHWCDomain);
+      [[maybe_unused]] auto new_node_ref =
+        onnx_layout_transformation::SwapNodeOpTypeAndDomain(*api_graph, *node, node->OpType(), kMSInternalNHWCDomain);
```

Comment:
typically we'd add
```#else 
ORT_UNUSED_PARAMETER(new_node_ref);
```

but if all the builds pass with maybe_unused that seems fine as well. only difference would be that if the value becomes unused in the `#if` as well that wouldn't be noticed. probably doesn't matter though.